hackathon_site/dashboard/frontend/src/components/cart/CartSummary/CartSummary.test.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import { render, waitFor } from "@testing-library/react";
+import CartSummary from "components/cart/CartSummary/CartSummary";
+import { cartQuantity, mockCartItems } from "testing/mockData";
+import store, { makeStore, RootState } from "slices/store";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+import thunk, { ThunkDispatch } from "redux-thunk";
+import {
+    initialState,
+    cartReducerName,
+    submitOrder,
+    cartSliceSelector,
+    cartSelectors,
+} from "slices/hardware/cartSlice";
+import { post } from "api/api";
+import { displaySnackbar } from "slices/ui/uiSlice";
+import { makeMockApiListResponse } from "testing/utils";
+import { AnyAction } from "redux";
+
+it(`Renders correctly when it reads the number ${cartQuantity}`, async () => {
+    const store = makeStore();
+    const { getByText } = render(
+        <Provider store={store}>
+            <CartSummary cartQuantity={cartQuantity} />
+        </Provider>
+    );
+    await waitFor(() => {
+        expect(getByText(cartQuantity.toString())).toBeInTheDocument();
+    });
+});
+
+jest.mock("api/api", () => ({
+    ...jest.requireActual("api/api"),
+    post: jest.fn(),
+}));
+
+const mockedPost = post as jest.MockedFunction<typeof post>;
+
+const mockState: RootState = {
+    ...store.getState(),
+    [cartReducerName]: initialState,
+};
+
+type DispatchExts = ThunkDispatch<RootState, void, AnyAction>;
+const mockStore = configureStore<RootState, DispatchExts>([thunk]);
+
+describe("Selectors", () => {
+    test("cartSliceSelector returns the cart store", () => {
+        expect(cartSliceSelector(mockState)).toEqual(mockState[cartReducerName]);
+    });
+});
+
+describe("submitOrder Thunk and Reducer", () => {
```

Comment:
Same with this one, should go in cartslice.test.s

Reply:
Should I keep this test "Dispatches a snackbar on API failure", inside the describe block because its a ui test?

Reply:
no you can move that to the cartslice as well, UI tests are tests that interact with the screen and displaying a snackbar does show a ui component but you're not clicking or checking if anything is displayed on the screen.